there is some strangeness in the gh action on Windows, the builds just failed 

```
 1>Running cpp protocol buffer compiler on D:/a/zq/zq/proto/pingpong.proto
  The system cannot find the batch label specified - VCEnd
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(237,5): error MSB8066: Custom build for 'D:\a\zq\zq\build\msvc22\CMakeFiles\a47caf46abbf6d92bfb436489541e164\pingpong.pb.h.rule' exited with code 1. [D:\a\zq\zq\build\msvc22\proto\zqproto.vcxproj]
Error: Process completed with exit code 1.
```

impossible to reproduce, it worked before the change, and it works on local Windows 10 and Windows 11 installations, 
so proto from Windows is now disabled, (what tests also this feature )

lets see what we do with the Windows build in future

--- a/.github/workflows/build-all.yml
+++ b/.github/workflows/build-all.yml
@@ -22,6 +22,11 @@ jobs:
             vcpkgCommitId: '6e31ee33cc9fc93599c4ceb38e229098cf339bb7'
             cmake_preset: msvc22
             extra_args: -D ZQ_WITH_PROTO=OFF
+            # there was this odd error that could not be reproduced locally on Windos 10 or 11
+            #  -> Running cpp protocol buffer compiler on D:/a/zq/zq/proto/pingpong.proto
+            # The system cannot find the batch label specified - VCEnd
+            # error MSB8066
+            # need to look into it extra ,but for now, disable proto on Windows
           - os: ubuntu-latest
             vcpkgCommitId: '6e31ee33cc9fc93599c4ceb38e229098cf339bb7'
             cmake_preset: ninja
@@ -48,12 +53,12 @@ jobs:
           cmakeVersion: "~3.28.0"  # <--= optional, use most recent 3.25.x version
           ninjaVersion: "^1.12.0"  # <--= optional, use most recent 1.x version
 
-      # - name: Dump the content of $RUNNER_TEMP
-      #   run: find $RUNNER_TEMP
-      #   shell: bash
-      # - name: Dump the content of $RUNNER_WORKSPACE
-      #   run: find $RUNNER_WORKSPACE
-      #   shell: bash
+      # it's not good to not test proto on Windows, but that accelerates the build for Windows
+      - name: Remove protobuf from vcpkg.json on Windows
+        run: |
+          (Get-Content -path vcpkg.json) | Where-Object { $_ -notmatch '^\s*"protobuf",\s*$' } | Set-Content -Path vcpkg.json
+        shell: pwsh
+        if: matrix.os == 'windows-latest'
 
       - name: Setup vcpkg
         uses: lukka/run-vcpkg@v11
@@ -65,9 +70,6 @@ jobs:
           vcpkgGitCommitId: '${{ matrix.vcpkgCommitId}}'
           # The vcpkg.json file, which will be part of cache key computation.
           vcpkgJsonGlob: 'vcpkg.json'
-
-      # - name: Prints output of run-vcpkg's action
-      #   run: echo "root='${{ steps.runvcpkg.outputs.RUNVCPKG_VCPKG_ROOT_OUT }}', triplet='${{ steps.runvcpkg.outputs.RUNVCPKG_VCPKG_DEFAULT_TRIPLET_OUT }}'"
 
       - name: Configure CMake
         run: |

--- a/.github/workflows/build-all.yml
+++ b/.github/workflows/build-all.yml
@@ -21,7 +21,7 @@ jobs:
           - os: windows-latest
             vcpkgCommitId: '6e31ee33cc9fc93599c4ceb38e229098cf339bb7'
             cmake_preset: msvc22
-            extra_args: -D ZQ_WITH_PROTO=OFF
+            extra_args: -D ZQ_WITH_PROTO=ON
             # there was this odd error that could not be reproduced locally on Windos 10 or 11
             #  -> Running cpp protocol buffer compiler on D:/a/zq/zq/proto/pingpong.proto
             # The system cannot find the batch label specified - VCEnd
@@ -41,7 +41,7 @@ jobs:
         if: matrix.os == 'ubuntu-latest'
         run: |
           sudo apt-get update
-          sudo apt-get install -y cmake libtool autoconf automake lcov
+          sudo apt-get install -y libtool autoconf automake lcov
 
       - name: Install dependencies on Mac
         if: matrix.os == 'macos-latest'
@@ -54,11 +54,11 @@ jobs:
           ninjaVersion: "^1.12.0"  # <--= optional, use most recent 1.x version
 
       # it's not good to not test proto on Windows, but that accelerates the build for Windows
-      - name: Remove protobuf from vcpkg.json on Windows
-        run: |
-          (Get-Content -path vcpkg.json) | Where-Object { $_ -notmatch '^\s*"protobuf",\s*$' } | Set-Content -Path vcpkg.json
-        shell: pwsh
-        if: matrix.os == 'windows-latest'
+      # - name: Remove protobuf from vcpkg.json on Windows
+      #   run: |
+      #     (Get-Content -path vcpkg.json) | Where-Object { $_ -notmatch '^\s*"protobuf",\s*$' } | Set-Content -Path vcpkg.json
+      #   shell: pwsh
+      #   if: matrix.os == 'windows-latest'
 
       - name: Setup vcpkg
         uses: lukka/run-vcpkg@v11
@@ -73,7 +73,8 @@ jobs:
 
       - name: Configure CMake
         run: |
-          cmake --preset ${{ matrix.cmake_preset }}  ${{ matrix.extra_args }}
+          cmake --preset ${{ matrix.cmake_preset }}
+#          cmake --preset ${{ matrix.cmake_preset }} ${{ matrix.extra_args }}
 
       - name: Build
         run: |

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,7 +41,7 @@ find_package(astr CONFIG REQUIRED)
 find_package(ZeroMQ CONFIG REQUIRED)
 find_package(tl-expected CONFIG REQUIRED)
 find_package(fmt CONFIG REQUIRED) # until apple clang catches up
-find_package(Protobuf CONFIG REQUIRED)
+
 
 list(PREPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 include(setup_compiler)
@@ -74,6 +74,7 @@ target_include_directories(zq INTERFACE
 
 
 if (ZQ_WITH_PROTO)
+    find_package(Protobuf CONFIG REQUIRED)
     target_compile_definitions(zq INTERFACE "ZQ_PROTO")
     add_subdirectory(proto)
 endif()

